spec-tests/src/test/java/org/candlepin/spec/imports/ImportWarningSpecTest.java

```diff
@@ -0,0 +1,266 @@
+        PoolDTO pool = expiredEnt.getPool();
+        pool.setEndDate(endDate);
+        expiredEnt.setPool(pool);
+
+        File modifiedExport = File.createTempFile(export.file().getName().replace(".zip", ""), ".zip");
+        modifiedExport.deleteOnExit();
+
+        // Create a copy of the export to replace the existing entitlement with the expired one.
+        try (ZipOutputStream zos = new ZipOutputStream(new FileOutputStream(modifiedExport))) {
+            try (ZipFile exportZip = new ZipFile(exportFile)) {
+                exportZip.stream().forEach(entry -> {
+                    if (entry.getName().equals("consumer_export.zip")) {
+                        File modifiedConsumerExport = copyConsumerExportAndReplaceEnt(export, expiredEnt);
+
+                        // Copy over the modified consumer_export.zip
+                        transfer(entry, modifiedConsumerExport, zos);
+                    }
+                    else {
+                        // Copy the signature file
+                        transfer(exportZip, entry, zos);
+                    }
+                });
+            }
+        }
+
+        // verify we get a warning
+        importJob = adminClient.owners().importManifestAsync(owner.getKey(), List.of("SIGNATURE_CONFLICT", "MANIFEST_SAME"), modifiedExport);
+        importJob = adminClient.jobs().waitForJob(importJob);
+        assertThat(importJob)
+            .returns("FINISHED", AsyncJobStatusDTO::getState)
+            .extracting(AsyncJobStatusDTO::getResultData)
+            .asString()
+            .contains("SUCCESS_WITH_WARNING")
+            .contains(owner.getKey() + " file imported forcibly.One or more inactive subscriptions found in the file.");
+    }
+
+    @Test
+    public void shouldWarnAboutNoActiveSubscriptions() throws Exception {
+        ApiClient adminClient = ApiClients.admin();
+        Export export = new ExportGenerator(adminClient)
+            .minimal()
+            .export();
+
+        OwnerDTO owner = adminClient.owners().createOwner(Owners.random());
+
+        AsyncJobStatusDTO importJob = adminClient.owners().importManifestAsync(owner.getKey(), List.of(), export.file());
+        importJob = adminClient.jobs().waitForJob(importJob);
+        assertThat(importJob)
+            .returns("FINISHED", AsyncJobStatusDTO::getState)
+            .extracting(AsyncJobStatusDTO::getResultData)
+            .asString()
+            .contains("SUCCESS_WITH_WARNING")
+            .contains(owner.getKey() + " file imported successfully.No active subscriptions found in the file.");
+    }
+
+    /**
+     * Converts the provided entitlement to Json. Also ensures that date fields in the {@link EntitlementDTO}
+     * and nested objects will not run into deserialization issues.
+     *
+     * @param ent
+     *  the entitlement to convert to json
+     *
+     * @return
+     *  the bytes of the converted Json string
+     */
+    private byte[] getEntitlementJson(EntitlementDTO ent) {
```

Comment:
Why is this needed? Could we not just serialize it directly with the jackson API like `writeValueAsBytes`?

Reply:
I tried using the `writeValueAsByte` method on the ObjectMapper, but it looks like we still run into the issue where the date fields are converted into what looks like the epoch value `"created":1668437865`. This causes issues being deserialized on in Candlepin, including the error message below. The `getEntitlementJson` gets around this issue by using the toString of the OffsetDateTime as the created/updated values: `"created":"2022-11-14T14:50:27Z"`

Candlepin Error message:
```
org.candlepin.async.JobExecutionException: Unable to extract export archive
        at org.candlepin.async.tasks.ImportJob.execute(ImportJob.java:245)
        at org.candlepin.async.JobManager.executeJob(JobManager.java:1273)
        at org.candlepin.async.JobMessageReceiver$MessageListener.handleMessage(JobMessageReceiver.java:350)
        at org.candlepin.messaging.impl.artemis.ArtemisConsumer$ArtemisMessageForwarder.onMessage(ArtemisConsumer.java:59)
        at org.apache.activemq.artemis.core.client.impl.ClientConsumerImpl.callOnMessage(ClientConsumerImpl.java:1001)
        at org.apache.activemq.artemis.core.client.impl.ClientConsumerImpl$Runner.run(ClientConsumerImpl.java:1154)
        at org.apache.activemq.artemis.utils.actors.OrderedExecutor.doTask(OrderedExecutor.java:56)
        at org.apache.activemq.artemis.utils.actors.OrderedExecutor.doTask(OrderedExecutor.java:31)
        at org.apache.activemq.artemis.utils.actors.ProcessorBase.executePendingTasks(ProcessorBase.java:67)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at org.apache.activemq.artemis.utils.ActiveMQThreadFactory$1.run(ActiveMQThreadFactory.java:118)
Caused by: org.candlepin.sync.ImportExtractionException: Unable to extract export archive
        at org.candlepin.sync.Importer.doExport(Importer.java:452)
        at org.candlepin.sync.Importer.loadStoredExport(Importer.java:209)
        at org.candlepin.controller.ManifestManager.importStoredManifest(ManifestManager.java:220)
        at com.google.inject.persist.jpa.JpaLocalTxnInterceptor.invoke(JpaLocalTxnInterceptor.java:64)
        at org.candlepin.async.tasks.ImportJob.execute(ImportJob.java:229)
        ... 11 common frames omitted
Caused by: com.fasterxml.jackson.databind.exc.MismatchedInputException: Cannot deserialize value of type `java.util.Date` from Floating-point value (token `JsonToken.VALUE_NUMBER_FLOAT`)
 at [Source: (FileReader); line: 1, column: 12] (through reference chain: org.candlepin.dto.manifest.v1.EntitlementDTO["created"])

```

Reply:
Taking a closer look a this, we can disable the SerializationFeature.WRITE_DATES_AS_TIMESTAMPS on the ObjectMapper and that prevents the conversion of the OffsetDateTime to a timestamp value and allows use to use the `writeValueAsBytes` and get rid of the `getEntitlementJson` method!